DatadogSessionReplay/Sources/Recorder/RecordingCoordinator.swift

```diff
@@ -115,9 +123,26 @@ internal class RecordingCoordinator {
         srContextPublisher.setHasReplay(hasReplay)
     }
 
-    /// Captures the next recording if conditions are met.
+    private var lastTriggerDate: Date?
+
+    private var shouldSkipTrigger: Bool {
+        return dateProvider.now.timeIntervalSince(lastTriggerDate ?? .distantPast) < Constants.throttingRate
+    }
+
+    private func didTrigger() {
+        guard shouldSkipTrigger == false else {
+            return
+        }
+        lastTriggerDate = dateProvider.now
+        captureNextRecord()
+    }
+
     private func captureNextRecord() {
-        /// We don't capture any snapshots if the RUM context has no view ID.
+        guard isSampled == true && recordingEnabled == true else {
```

Comment:
It seems like this check isn't necessary since it’s already handled in `evaluateRecordingConditions`. If you’d like to keep it as an additional safety check, we could centralize this logic with a computed property:

```
private var shouldRecord: Bool {
    return isSampled && recordingEnabled
}
```